src/Agent.Worker/StepsRunner.cs

```diff
@@ -84,6 +85,48 @@ public sealed class StepsRunner : AgentService, IStepsRunner
                     step.ExecutionContext.Result = TaskResult.Failed;
                 }
 
+                // Wait till all async commands finish.
```

Comment:
I think this is fine for now. Per offline discussion, a more clean approach *might be* to rely on the token to determine whether the task result is canceled. Better to do that in a separate PR, so I think this is appropriate for now.